src/base/trafficGraph.js

```diff
@@ -371,6 +374,10 @@ class TrafficGraph extends EventEmitter {
   setState (state, force, parentState) {
     let updatedState = false;
     if (state && Object.keys(state).length > 0) {
+      // it's important to update entryNode when we're switching views
+      if (state.entryNode) {
+        this.entryNode = state.entryNode;
+      }
```

Comment:
Layout shouldn't be broken because it should just figure out entryNode if there is none specified...  The only thing layout uses entryNode for is to force rank promotions.